Equations for measures (#13)
MrLongbottom/P10-Article: sections/Experiment.tex

\subsection{Evaluation Metrics}\label{sec:experiment_metrics}
All models previously mentioned are evaluated on the following evaluation metrics.
The main metric used, in this experiment, is the topic coherence metric.

> This metric indicates how semantically similar the top words within each topics are and will be an indication of the quality of the topics a topic model generates~\cite{topic_coherence_2015}.

Reviewer: ```suggestion
This metric indicates how semantically similar the top words within each topic are, and will be an indication of the quality of the topics a topic model generates~\cite{topic_coherence_2015}.
```

> There are many different equation for calculating topic coherence, for this paper we will be using $C_v$ coherence.

Reviewer: ```suggestion
There are many different equations for calculating topic coherence, but for this paper we will be using $C_v$ coherence.
```

$C_v$ is calculated using the following steps, as presented by~\citet{Syed2017coherence}:
\begin{enumerate}
	\item Topic-word segmentation into word set pairs
	\item Word and word pair probability calculation
	\item Word set confirmation measure
	\item Aggregation of confirmation measures
\end{enumerate}
\begin{equation}
	\overrightarrow{v}(W') = \left\{ \sum_{w_i \in W'} \text{NPMI}(w_i, w_j)^\gamma \right\}_{j=1,\dots,|W|}
\end{equation}
\begin{equation}
	\text{NPMI}(w_i,w_j)^\gamma = \left( \frac
	{log\frac{P(w_i,w_j) + \epsilon}{P(w_i)\cdot P(w_j)}}
	{-log(P(w_i,w_j) + \epsilon)} \right)
\end{equation}
\begin{equation}
	\phi_{S_i}(\overrightarrow{u}, \overrightarrow{w}) = \frac
	{\sum_{i = 1}^{|W|} u_i \cdot w_i}
	{\|\overrightarrow{u}\|_2 \cdot \|\overrightarrow{w}\|_2}
\end{equation}
\begin{equation}
	C_v = \frac{1}{|S|}\sum_{|S|}^{i=1}\phi_{S_i}
\end{equation}

> \noindent where $W$ is a set of the top-N words in a topic, $S$ is a set of all possible pairs $(W',W*)$ where $W'\in W$ and $W* = W$. This is the segmentation.

Reviewer: ```suggestion
\noindent where $W$ is a set of the top-N words in a topic, and $S$ is a set of all possible pairs $(W',W*)$ where $W'\in W$ and $W* = W$.
This is the segmentation.
```

NPMI is the normalized pointwise mutual information.
$\overrightarrow{v}(W')$ is the context vector for $W'$ which is made by pairing them to all words in $W$.

> $phi_{S_i}$ is the confirmation measure for a segmentation pair $S_i = (W',W*)$, that describes how much $W*$ supports $W'$.

Reviewer: ```suggestion
$\phi_{S_i}$ is the confirmation measure for a segmentation pair $S_i = (W',W*)$, that describes how much $W*$ supports $W'$.
```

> It is calculated by the consine similarity of all context vectors $phi_{S_i}(\overrightarrow{u},\overrightarrow{w})$ in $S_i$, with $\overrightarrow{v}(W') \in \overleftarrow{u}$ and $\overrightarrow{v}(W*) \in \overrightarrow{w}$.

Reviewer: ```suggestion
It is calculated by the cosine similarity of all context vectors $\phi_{S_i}(\overrightarrow{u},\overrightarrow{w})$ in $S_i$, with $\overrightarrow{v}(W') \in \overrightarrow{u}$ and $\overrightarrow{v}(W*) \in \overrightarrow{w}$.
```
Hvis der er en grund til at der bruges `overleftarrow` jeg ikke kunne se, kan du bare ændre det tilbage.

> $\epsilon$ is used to avoid $log(0)$ and $\gamma$ is used to further weight high NPMI values higher.

Reviewer: ```suggestion
$\epsilon$ is used to avoid $log(0)$ and $\gamma$ is used to further weigh high NPMI values higher.
```

\subsection{Evaluation Metrics}\label{sec:experiment_metrics}
All models previously mentioned are evaluated on the following evaluation metrics.
The main metric used, in this experiment, is the topic coherence metric.
This metric indicates how semantically similar the top words within each topic are, and will be an indication of the quality of the topics a topic model generates~\cite{topic_coherence_2015}.

> There are many different equations for calculating topic coherence, but for this paper we will be using $C_v$ coherence.

Reviewer 2: ```suggestion
There are many different equations for calculating topic coherence, but for this paper, we will be using $C_v$ coherence.
```

Topic difference is another metric that is used to check the quality of the topic model.
It is based on the assumption that a good topic model will have little overlap between topics.
It is therefore not necessarily the best measure of the final quality of a topic model, but it can show potential problems with a model.
\begin{equation}
	TopicDifference = \frac{1}{K \cdot K} \sum_{i=1}^{K} \sum_{j=1}^{K} JS(\beta_{i},\beta_{j})
\end{equation}

> \noindent where $JS$ is jenson-shannon distance, $K$ is the number of topics, and $\beta_{k}$ is the topic-word distribution for topic $k$.

Reviewer: ```suggestion
\noindent where $JS$ is Jensen-Shannon distance, $K$ is the number of topics, and $\beta_{k}$ is the topic-word distribution for topic $k$.
```

Reviewer 2: måske lav jenson shannon til gls
